rama-cli/src/cmd/stunnel/mod.rs

```diff
@@ -0,0 +1,250 @@
+                    .map_err(|e| format!("Failed to read key file: {e}"))?;
+
+                ServerConfig::new(ServerAuth::Single(ServerAuthData {
+                    cert_chain: DataEncoding::Pem(cert_data),
+                    private_key: DataEncoding::Pem(key_data),
+                    ocsp: None,
+                }))
+            } else {
+                tracing::info!("Using auto-generated self-signed certificate");
+                new_server_config(None)
+            };
+
+            let acceptor_data = TlsAcceptorData::try_from(server_config)?;
+
+            let listener = TcpListener::bind(args.listen.clone())
+                .await
+                .expect("Failed to bind stunnel server");
+
+            let listen_addr = args.listen.clone();
+            let forward_addr = args.forward;
+
+            shutdown.spawn_task_fn(async move |guard| {
+                tracing::info!("Stunnel server is running...");
+                tracing::info!(
+                    "Listening on {} and forwarding to {}",
+                    listen_addr,
+                    forward_addr
+                );
+                let tcp_service =
+                    TlsAcceptorLayer::new(acceptor_data).layer(Forwarder::new(forward_addr));
+                listener.serve_graceful(guard, tcp_service).await;
+            });
+
+            shutdown
+                .shutdown_with_limit(Duration::from_secs(30))
+                .await
+                .expect("graceful shutdown");
+
+            Ok(())
+        }
+
+        StunnelSubcommand::Client(args) => {
+            let shutdown = Shutdown::default();
+
+            let mut tls_builder = TlsConnectorDataBuilder::new();
+
+            // Configure certificate verification
+            if args.insecure {
+                tls_builder = tls_builder.with_server_verify_mode(ServerVerifyMode::Disable);
+                tracing::warn!("TLS certificate verification disabled (--insecure flag)");
+                tracing::warn!("This is insecure and should only be used for testing!");
+            } else if let Some(cacert_path) = &args.cacert {
+                tracing::info!(
+                    cacert.path = ?cacert_path,
+                    "Loading CA certificate for server verification"
+                );
+
+                let ca_pem = std::fs::read_to_string(cacert_path)
+                    .map_err(|e| format!("Failed to read CA certificate file: {e}"))?;
+
+                let ca_cert = X509::from_pem(ca_pem.as_bytes())
+                    .map_err(|e| format!("Failed to parse CA certificate: {e}"))?;
+
+                let mut store_builder = X509StoreBuilder::new()
+                    .map_err(|e| format!("Failed to create certificate store: {e}"))?;
+                store_builder
+                    .add_cert(ca_cert)
+                    .map_err(|e| format!("Failed to add CA certificate to store: {e}"))?;
+
+                tls_builder =
```

Comment:
Doesn't this have something like `set_server_verify_cert_store`??? so that you do not need to re-assign it